front/cypress/e2e/admin/users/seat-billing.cy.ts

```diff
@@ -0,0 +1,708 @@
+          lastName: user3LastName,
+          email: user3Email,
+          password: user3Password,
+        },
+        {
+          firstName: user4FirstName,
+          lastName: user4LastName,
+          email: user4Email,
+          password: user4Password,
+        },
+        {
+          firstName: user5FirstName,
+          lastName: user5LastName,
+          email: user5Email,
+          password: user5Password,
+        },
+        {
+          firstName: user6FirstName,
+          lastName: user6LastName,
+          email: user6Email,
+          password: user6Password,
+        },
+        {
+          firstName: user7FirstName,
+          lastName: user7LastName,
+          email: user7Email,
+          password: user7Password,
+        },
+      ]);
+    });
+
+    beforeEach(() => {
+      cy.setAdminLoginCookie();
+    });
+
+    after(() => {
+      cleanUp();
+      cy.apiRemoveProject(projectId);
+    });
+
+    it('allows user to add a moderator and shows a confirmation when needed', () => {
+      cy.visit(`admin/projects/${projectId}/permissions`);
+      cy.acceptCookies();
+
+      cy.intercept(`**/projects/${projectId}/moderators`).as(
+        'moderatorsRequest'
+      );
+      cy.wait('@moderatorsRequest');
+
+      // Add moderator and check that they are shown in the list
+      cy.get('#projectModeratorUserSearch').should('exist');
+      cy.get('#projectModeratorUserSearch').type(`${user2Email}`);
+      cy.get(`[data-cy="e2e-user-${user2Email}"]`).click();
+      cy.get('[data-cy="e2e-add-project-moderator-button"]').click();
+      testShowModalOnAddingModerator();
+      cy.get('.e2e-admin-list').contains(user2Email);
+
+      // Add moderator and check that they are shown in the list
+      cy.get('#projectModeratorUserSearch').should('exist');
+      cy.get('#projectModeratorUserSearch').type(`${user3Email}`);
+      cy.get(`[data-cy="e2e-user-${user3Email}"]`).click();
+      cy.get('[data-cy="e2e-add-project-moderator-button"]').click();
+      testShowModalOnAddingModerator();
+      cy.get('.e2e-admin-list').contains(user3Email);
+
+      // Add moderator and check that they are shown in the list
+      cy.get('#projectModeratorUserSearch').should('exist');
+      cy.get('#projectModeratorUserSearch').type(`${user4Email}`);
+      cy.get(`[data-cy="e2e-user-${user4Email}"]`).click();
+      cy.get('[data-cy="e2e-add-project-moderator-button"]').click();
+      testShowModalOnAddingModerator();
+      cy.get('.e2e-admin-list').contains(user4Email);
+
+      // Add moderator and check that they are shown in the list
+      cy.get('#projectModeratorUserSearch').should('exist');
+      cy.get('#projectModeratorUserSearch').type(`${user5Email}`);
+      cy.get(`[data-cy="e2e-user-${user5Email}"]`).click();
+      cy.get('[data-cy="e2e-add-project-moderator-button"]').click();
+      testShowModalOnAddingModerator();
+      cy.get('.e2e-admin-list').contains(user5Email);
+    });
```

Comment:
> Wouldn't it be much easier to already set the data so that we're 1 click away from the confirmation modal rather than having to perform this action 4x?

We can't do that because we can't control the data fully in this case. See previous explanations. We would have to control all the roles of the users in the app which is not practical since different tests could leave the state with more data and users. See dangling state cypress